pyvisa_py/tcpip.py

```diff
@@ -64,8 +71,48 @@ class TCPIPInstrSession(Session):
 
     @staticmethod
     def list_resources() -> List[str]:
-        # TODO: is there a way to get this?
-        return []
+        broadcast_addr = []
+        if psutil is not None:
+            # Get broadcast address for each interface
+            for interface, snics in psutil.net_if_addrs().items():
+                for snic in snics:
+                    if snic.family is socket.AF_INET:
+                        addr = snic.address
+                        mask = snic.netmask
+                        network = ipaddress.IPv4Network(addr + "/" + mask, strict=False)
+                        broadcast_addr.append(str(network.broadcast_address))
+        else:
+            msg = "To discover devices on all interfaces, please install psutil"
+            Session.register_unavailable(constants.InterfaceType.tcpip, "INSTR", msg)
```

Comment:
No doing this make TCPIP resources unavailable without psutil !

Reply:
Got it, wasn't sure if I should have added this or not. The default behaviour if `psutil` is not found is now to fail silently, and only broadcast on the default interface.